x/username/username.go

```diff
@@ -0,0 +1,73 @@
+package username
+
+import (
+	"encoding/json"
+	"regexp"
+
+	"github.com/iov-one/weave/errors"
+)
+
+// Username represents a name registered on certain domain. A valid username is
+// in format <name>*<domain>. Username is case sensitive.
+type Username string
+
+// ParseUsername returns a valid username instance, extracted from given string
+// representation.
+func ParseUsername(s string) (Username, error) {
+	u := Username(s)
+	return u, u.Validate()
+}
+
+// Name returns the name part. This is the value before the separator.
+func (u Username) Name() string {
+	for i, c := range u {
+		if c == '*' {
+			return string(u[:i])
+		}
+	}
+	return ""
+}
+
+// Name returns the name part. This is the value before the separator.
+func (u Username) Domain() string {
+	for i, c := range u {
+		if c == '*' {
+			return string(u[i+1:])
+		}
+	}
+	return ""
+}
+
+// Bytes returns the byte representation of the username. Use this method when
+// you need to use a username value as a database key.
+func (u Username) Bytes() []byte {
+	return []byte(u)
+}
+
+func (u Username) String() string {
+	return string(u)
+}
+
+func (u Username) Validate() error {
+	if !validUsername(string(u)) {
+		return errors.Wrap(errors.ErrInput, "invalid username")
+	}
+	return nil
+}
+
+var validUsername = regexp.MustCompile(`\w{3,}\*\w{3,}`).MatchString
```

Comment:
Current username requirements are here 
https://github.com/iov-one/weave/blob/master/cmd/bnsd/x/nft/username/msg.go#L20
 (it would be good to keep those rules for the part before the *)

Reply:
Addressed in https://github.com/iov-one/weave/pull/771/commits/65b0676f0ab27e05440c78d7ab7dda58f44c73cc